File: 프로그래머스/3/43162. 네트워크/네트워크.py
def bfs(n, start, visited, computers):
    queue = [start]
    visited[start] = 1
    while queue:
        num = queue.pop(0)
        for i in range(n):
            if computers[num][i] == 1 and visited[i] == 0:
                queue.append(i)
                visited[i] = 1
    return visited


def solution(n, computers):
    visited = [0] * n
    cnt = 0
    while sum(visited) != n:
        for i in range(n):
            if visited[i] == 0:
                visited = bfs(n, i, visited, computers)
                cnt += 1
                break

    return cnt

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] computers) {
        int[] visited = new int[n];
        Queue<Integer> queue = new LinkedList<>();

        int cnt = 0;

        for (int i = 0; i < n; i++) {
            if (visited[i] == 1) {
                continue;
            }
            queue.add(i);
            visited[i] = 1;
            cnt++;
            while (!queue.isEmpty()) {
                int x = queue.poll();
                for (int j = 0; j < n; j++) {
                    // 연결되어 있고 방문하지 않았을 경우
                    if (computers[x][j] == 1 && visited[j] == 0) {
                        queue.add(j);
                        visited[j] = 1;
                    }
                }
            }
        }

        int answer = cnt;
        return answer;
    }
}